src/lib/storage/reference_column/reference_column_iterable.hpp

```diff
@@ -55,14 +60,17 @@ class ReferenceColumnIterable : public ColumnIterable<ReferenceColumnIterable<T>
       const auto chunk_id = _pos_list_it->chunk_id;
       const auto& chunk_offset = _pos_list_it->chunk_offset;
 
-      const auto chunk = _table->get_chunk(chunk_id);
-      const auto column = chunk->get_column(_column_id);
+      if (chunk_id != _cached_chunk_id || !_cached_column) {
```

Comment:
If you set _cached_chunk_id to be `INVALID_CHUNK_ID` initially you could get rid of the `!_cached_column` bit - though branch prediction probably already covers that optimization.